commons-web/src/main/java/eu/europeana/api/commons/service/authorization/BaseAuthorizationService.java

```diff
@@ -163,6 +167,53 @@ private boolean isOperationAuthorized(String operation, List<GrantedAuthority> a
 	return false;
     }
 
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(String etag, HttpServletRequest request) throws HttpException {
+
+		String ifMatchHeader = request.getHeader(HttpHeaders.IF_MATCH);
+		if (ifMatchHeader != null) {
+		    try {
+				if (!etag.equals(ifMatchHeader))
+				    throw new HeaderValidationException(I18nConstants.INVALID_PARAM_VALUE, HttpHeaders.IF_MATCH,
+					    ifMatchHeader);
+		    } catch (NumberFormatException e) {
+				throw new HeaderValidationException(I18nConstants.INVALID_PARAM_VALUE, HttpHeaders.IF_MATCH,
+					ifMatchHeader);
+		    }
+		}
+    }
+    
+    /**
+     * This method generates etag for response header.
+     * 
+     * @param timestamp The date of the last modification
+     * @param format       The MIME format
+     * @param version      The API version
+     * @return etag value
+     */
+    public String generateETag(Date timestamp, String format, String version) {
```

Comment:
this method must not be implemented in authorization service, it belogns to BaseRestControler

Reply:
moved to BaseRestController